tests/manage/pv_services/space_reclaim/test_auto_reclaim_space_cronjob.py

```diff
@@ -0,0 +1,223 @@
+
+            for temp_file in self.temp_files_list:
+                if os.path.exists(temp_file):
+                    run_cmd(f"rm {temp_file}")
+
+            run_oc_command(cmd=f"delete namespace {self.namespace}")
+
+        request.addfinalizer(finalizer)
+
+    def test_reclaim_space_cronjob(self):
+        """
+        Test case to check reclaim space cronjobs are created correctly for rbd pvcs in openshift-* namespace
+        """
+        num_of_pvcs = 10
+        namespace = f"openshift-{uuid4().hex}"
+        self.namespace = namespace
+        result = run_oc_command(cmd=f"create namespace {self.namespace}")
+        assert ERRMSG not in result[0], (
+            f"Failed to create namespace with name {namespace}" f"got result: {result}"
+        )
+        logger.info(f"Namespace {namespace} created")
+
+        result = run_oc_command(cmd=f"get namespace {namespace} -o yaml")
+        namespace_dict = yaml.safe_load("\n".join(result))
+
+        schedule = namespace_dict["metadata"]["annotations"][
+            "reclaimspace.csiaddons.openshift.io/schedule"
+        ]
+        assert (
+            schedule == "@weekly"
+        ), f"Namespace {namespace} created with schedule {schedule}, expected @weekly"
+        logger.info(f"Existence of schedule {schedule} validated.")
+
+        self.pvc_objs_created, _ = helpers.create_multiple_pvcs(
+            sc_name=constants.DEFAULT_STORAGECLASS_RBD,
+            namespace=self.namespace,
+            number_of_pvc=num_of_pvcs,
+            size="1Gi",
+        )
+
+        logger.info("Wait for all of the PVCs to be in Bound state")
+        performance_lib.wait_for_resource_bulk_status(
+            "pvc",
+            num_of_pvcs,
+            self.namespace,
+            constants.STATUS_BOUND,
+            num_of_pvcs * 2,
+            5,
+        )
+        # in case of creation faliure, the wait_for_resource_bulk_status function
+        # will raise an exception. so in this point the creation succeed
+
+        result = self.wait_for_cronjobs(
+            True, f"No reclaim space cron jobs exist in namespace {namespace}"
+        )
+        logger.info(f"Reclaim space jobs after PVC creation {result}")
+
+        cronjob_names = []
+        for cronjobs in result[1:]:
+            cronjob_names.append(cronjobs.split()[0])
+
+        for pvc_obj in self.pvc_objs_created:
+            cronjob_for_pvc = [
+                name for name in cronjob_names if name.startswith(f"{pvc_obj.name}-")
+            ]
+            assert (
+                len(cronjob_for_pvc) == 1
+            ), f"Expected exactly one reclaim space cron job for  pvc {pvc_obj.name}, {len(cronjob_for_pvc)} found."
+        logger.info("Existence of reclaim space cron jobs for RBD PVCs was validated.")
+        for pvc_obj in self.pvc_objs_created:
+            pvc_obj.delete()
+
+        self.wait_for_cronjobs(
+            False, "After PVCs deletion some reclaimspacecronjobs were left"
+        )
```

Comment:
Please consider adding a log message about this validation. It would be beneficial for understanding the test flow and also during debugging the test failures

Reply:
Added, thank you!